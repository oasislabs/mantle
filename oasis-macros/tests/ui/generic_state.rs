#![feature(proc_macro_hygiene)]
#[oasis_std::contract]
mod contract {
    #[derive(Contract, Default)]
    pub struct State<T>(Option<T>);

    impl<T: Default> State<T> {
        pub fn new(ctx: &Context) -> Result<Self> {
            Ok(Default::default())
        }

        fn hmmm() -> Result<()> {
            Err(failure::format_err!("hmm"))
        }
    }
}

fn main() {}
