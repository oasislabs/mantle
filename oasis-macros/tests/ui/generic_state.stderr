error: Contract cannot contain generic types.
 --> $DIR/generic_state.rs:5:21
  |
5 |     pub struct State<T>(Option<T>);
  |                     ^

error: aborting due to previous error

